libr/bin/bin.c

```diff
@@ -1062,16 +1061,17 @@ R_API void r_bin_list_archs(RBin *bin, int mode) {
 			pj_kn (pj, "offset", boffset);
 			pj_kn (pj, "size", obj_size);
 			if (!strcmp (arch, "mips")) {
-				pj_ks (pj, "isa", h_flag);
+				pj_ks (pj, "isa", info->cpu);
+				pj_ks (pj, "features", info->features);
 			}
 			if (machine) {
 				pj_ks (pj, "machine", machine);
 			}
 			pj_end (pj);
 			break;
 		default:
-			str_fmt = strcmp (h_flag, "unknown_flag")? sdb_fmt ("%s_%i %s", arch, bits, h_flag) \
-					  : sdb_fmt ("%s_%i", arch, bits);
+			str_fmt = strcmp (arch, "mips")?sdb_fmt ("%s_%i", arch, bits) \
```

Comment:
Why wasn't it ok as it was before? I think the check before was a bit more generic. It is not good, if possible, to have special handling for each arch. Doing `strcmp(arch, "mips")` makes this code arch-specific